src/main/java/com/Elessar/app/client/MyClient.java

```diff
@@ -0,0 +1,63 @@
+package com.Elessar.app.client;
+
+
+import com.Elessar.proto.Registration.RegistrationResponse;
+import com.Elessar.proto.Registration.RegistrationRequest;
+import com.google.api.client.http.*;
+import com.google.api.client.http.javanet.NetHttpTransport;
+import com.google.api.client.http.protobuf.ProtoHttpContent;
+import java.io.IOException;
+import java.net.URL;
+
+/**
+ * Created by Hans on 1/13/19.
+ */
+public class MyClient {
+    private final URL hostURL;
+
+    public MyClient(URL hostURL) {
+        this.hostURL = hostURL;
+    }
+
+    public void register (String userName, String password, String email, String phoneNumber) {
+        final RegistrationRequest.Builder user = RegistrationRequest.newBuilder();
+        user.setName(userName);
+        user.setPassword(password);
+        user.setEmail(email);
+        if (phoneNumber != null) {
+            user.setPhoneNumber(phoneNumber);
+        }
+
+        final HttpTransport HTTP_TRANSPORT = new NetHttpTransport();
+        final HttpRequestFactory REQ_FACTORY = HTTP_TRANSPORT.createRequestFactory();
+        try {
+            final URL regURL = new URL(hostURL.toString() + "/register");
+            final GenericUrl endURL = new GenericUrl(regURL);
+            final HttpContent content = new ProtoHttpContent(user.build());
+            final HttpRequest postRequest = REQ_FACTORY.buildPostRequest(endURL, content);
+            final HttpResponse postResponse = postRequest.execute();
+            RegistrationResponse regResponse = RegistrationResponse.parseFrom(postResponse.getContent());
+
+            if (regResponse.getSuccess()) {
+                System.out.println("User is Successfully Registered !");
+            } else {
+                System.out.println("User Registration Failed Because: " + regResponse.getFailReason());
+            }
+        } catch (IOException e) {
+            System.out.println("Cannot Execute Request !");
+        }
+    }
+
+    public void echo () {
+        final HttpTransport HTTP_TRANSPORT = new NetHttpTransport();
+        final HttpRequestFactory REQ_FACTORY = HTTP_TRANSPORT.createRequestFactory();
+        try {
+            final URL echoURL = new URL(hostURL.toString() + "/echo");
```

Comment:
have you tried to pass some string data as part of the echo request?

Reply:
Yes, I tried using `REQ_FACTORY.buildRequest(String "GET", GenericUrl url, HttpContent content)` and put a string into content, but it errors out saying cannot having non-zero content with GET method. So I removed the content for test

Reply:
Ok, based on this: https://stackoverflow.com/questions/978061/http-get-with-request-body
Seems it's possible but we should avoid it